the-collab-lab/tcl-69-smart-shopping-list
# [MF] styling list and items pages

## Description


This code changes a lot of things within the code. I will list them here to help guide you through this PR:

- In all directories with barrel files add all of the exports to these files.
- Goes through and uses the barrel files to consolidates imports
- Creates onClick handleSignIn function that navigates user to /list after sign in
- Builds out a custom CSS checkbox with a purple check mark that bounces in upon clicking
- Adds class names and styling to corresponding style sheets to apply global colors
- Moves delete list button outside of navigation link to /manage-list, leaving user on list page post delete
- Conditionally renders the categories in `SortedItemsMap`. If user doesn't have an item the category will not crowd UI
- Creates default input styling in index.css to globally style text inputs with these class names
- Adds temporary profile image on landing page post login. (this will be removed in merge conflict resolution)
- Removes unused css from Link.css leftover from refactoring 
-Adds purple check mark favicon (this can be reverted or changed)

## Related Issue

subissue #14 
closes #89 
closes #84 

## Acceptance Criteria

- [x] Ensure the list page and Manage list page adhere to app colors and global styling.
- [x] Refine search and create inputs by eliminating default styling and paint with app colors.
- [x] Update redirects post sign-in and delete list to navigate users to the lists page instead of the landing page or keep users on the lists page after deleting a list
- [x] standardize import structure across files for React, app functions/props, and CSS, ensuring index.js declarations are used for exports.

## Type of Changes

Use one or more labels to help your team understand the nature of the change(s) you’re proposing. E.g., `bug fix` or `enhancement` are common ones.

## Updates

### Before

#### Landing Page

![Screenshot 2024-04-06 at 12 21 36 AM](https://github.com/the-collab-lab/tcl-69-smart-shopping-list/assets/117051341/ec11e9de-3f54-4349-9a8f-99e87eb538ee)

#### Lists Page

![Screenshot 2024-04-06 at 12 23 23 AM](https://github.com/the-collab-lab/tcl-69-smart-shopping-list/assets/117051341/b18e21e1-b50c-4673-b875-c0727e590c78)

#### Manage-List Page

![Screenshot 2024-04-06 at 12 23 45 AM](https://github.com/the-collab-lab/tcl-69-smart-shopping-list/assets/117051341/4e403d53-756d-4a64-b868-c4ea3ff464f9)

#### Favicon

![Screenshot 2024-04-06 at 12 26 32 AM](https://github.com/the-collab-lab/tcl-69-smart-shopping-list/assets/117051341/0eaff5fd-8fa2-4089-b744-13561695f1e2)

### After

#### Landing Page

![Screenshot 2024-04-06 at 12 24 58 AM](https://github.com/the-collab-lab/tcl-69-smart-shopping-list/assets/117051341/140d61d8-3761-4051-827c-fa09945c0ebb)

#### Lists Page

![Screenshot 2024-04-06 at 12 25 16 AM](https://github.com/the-collab-lab/tcl-69-smart-shopping-list/assets/117051341/b99dda66-cbd7-4d0f-9572-b61b6397d122)

#### Manage-List Page

![Screenshot 2024-04-06 at 12 25 34 AM](https://github.com/the-collab-lab/tcl-69-smart-shopping-list/assets/117051341/84360bab-18ed-44b4-904f-b302251cd117)

![Screenshot 2024-04-06 at 12 25 53 AM](https://github.com/the-collab-lab/tcl-69-smart-shopping-list/assets/117051341/dd55fbb1-c777-43c5-aacc-98cc723b6fc3)

#### Favicon

![Screenshot 2024-04-06 at 12 24 14 AM](https://github.com/the-collab-lab/tcl-69-smart-shopping-list/assets/117051341/25e199c8-c825-4c25-89f4-51211104db6b)

## Testing Steps / QA Criteria

Login to app and start navigating around to see changes. Look over the AC and Summary to see all of the little details in the app.
